src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -31,6 +31,20 @@ protected CompareExchangeCommandBase(string key, long index, JsonOperationContex
 
         public abstract (long Index, object Value) Execute(TransactionOperationContext context, Table items, long index);
 
+        public unsafe bool Validate(TransactionOperationContext context, Table items, long index)
+        {
+            var dbKey = Key.ToLowerInvariant();
+            using (Slice.From(context.Allocator, dbKey, out Slice keySlice))
+            {
+                if (items.ReadByKey(keySlice, out var reader))
+                {
+                    var itemIndex = *(long*)reader.Read((int)ClusterStateMachine.UniqueItems.Index, out var _);
+                    return Index == itemIndex;
+                }
+            }
```

Comment:
You are missing handling for missing value.
Better to use the `Slice` ability to do lower casing, instead.